command/sub8_missions/sub8/sub_singleton.py

```diff
@@ -144,12 +172,18 @@ def check_goal(self):
             print "GOAL TOO HIGH"
             self._pos.position = -0.6
 
-    def go(self, *args, **kwargs):
+    def go(self, check_feasibility=True, *args, **kwargs):
+        '''
+        Set the sub's waypoint to _pose
+        @param check_feasibility Check that goal is 'feasibile' and correct otherwise, see check_goal
+        @return Defer object for action client to complete this move. yield on this to wait for move to complete
+        '''
         if self.print_only:
             print self._pose
             return self._sub.nh.sleep(0.1)
 
-        self.check_goal()
+        if check_feasibility:
```

Comment:
I think this is pretty much the same as `blind` isn't it? I think that'd be a good keyword to piggyback off of